neo/Ledger/Blockchain.cs

```diff
@@ -100,7 +99,7 @@ public Blockchain(NeoSystem system, Store store)
             this.system = system;
             this.MemPool = new MemoryPool(system, ProtocolSettings.Default.MemoryPoolMaxTransactions);
             this.Store = store;
-            lock (lockObj)
+            lock (typeof(Blockchain))
```

Comment:
This doesn't seem to be recommended:
https://bytes.com/topic/c-sharp/answers/249277-dont-lock-type-objects

I'm not a c# expert, but is this change really needed?

Reply:
We can change it, but one object in memory for only once, seems weird to me

Reply:
The author is very clear about it: `This practice is not acceptable, let alone
recommended.`

Reply:
Done! if is bad is bad :)